src/shogun/machine/gp/SingleSparseInference.cpp

```diff
@@ -112,8 +111,10 @@ class SingleSparseInferenceCostFunction: public FirstOrderBoundConstraintsCostFu
         void init()
 	{
 		m_obj=NULL;
-		SG_ADD((CSGObject **)&m_obj, "CSigleSparseInference__m_obj",
-			"m_obj in AdamUpdater", MS_NOT_AVAILABLE);
+		//The existing implementation in CSGObject::get_parameter_incremental_hash()
```

Comment:
```

void CSGObject::get_parameter_incremental_hash(uint32_t& hash, uint32_t& carry,
		uint32_t& total_length)
{
	for (index_t i=0; i<m_parameters->get_num_parameters(); i++)
	{
		TParameter* p=m_parameters->get_parameter(i);

		SG_DEBUG("Updating hash for parameter %s.%s\n", get_name(), p->m_name);

		if (p->m_datatype.m_ptype == PT_SGOBJECT)
		{
			if (p->m_datatype.m_ctype == CT_SCALAR)
			{
				CSGObject* child = *((CSGObject**)(p->m_parameter));

				if (child)
				{
					child->get_parameter_incremental_hash(hash, carry,
							total_length);
				}
			}
			else if (p->m_datatype.m_ctype==CT_VECTOR ||
					p->m_datatype.m_ctype==CT_SGVECTOR)
			{
				CSGObject** child=(*(CSGObject***)(p->m_parameter));

				for (index_t j=0; j<*(p->m_datatype.m_length_y); j++)
				{
					if (child[j])
					{
						child[j]->get_parameter_incremental_hash(hash, carry,
								total_length);
					}
				}
			}
		}
		else
			p->get_incremental_hash(hash, carry, total_length);
	}
}
```
The existing implementation is about tree traversal. It cannot handle loop. Please feel free to use graph traversal.